File: python/clock/clock.py
class Clock():
    def __init__(self, hour, minutes):
        self.hour = self.set_hour(hour)
        self.minutes = self.set_minutes(minutes)

    def set_hour(self, hour):
        return (hour - 24) % 24

    def set_minutes(self, minutes):
        self.hour += minutes / 60
        self.hour = self.set_hour(self.hour)
        return (60 + minutes) % 60

    def __str__(self):
        return "{:02d}:{:02d}".format(self.hour, self.minutes)

    def add(self, minutes):
        self.minutes += minutes
        self.minutes = self.set_minutes(self.minutes)
        return self

    def __eq__(clock1, clock2):
        if clock1.minutes == clock2.minutes and clock1.hour == clock2.hour:
            return True
        return False

File: ruby/clock/clock.rb
class Clock
  attr_accessor :time

  def initialize(hour, minutes = 0)
    @time = hour * 60 + minutes
  end

  def self.at(hour, minutes = 0)
    new(hour, minutes)
  end

  def to_s
    the_hour = time / 60
    the_minutes = time % 60
    if the_hour > 23
      the_hour -= 24
    elsif the_hour < 0
      the_hour += 24
    end
    "#{"%02d" % the_hour}:#{"%02d" % the_minutes}"
  end

  def +(minutes_to_add)
    self.time += minutes_to_add
    self
  end

  def -(minutes_to_subtract)
    self.time -= minutes_to_subtract
    self
  end

  def ==(clock2)
    to_s == clock2.to_s
  end
end
